include/coreir/ir/typecache.h

```diff
@@ -3,6 +3,8 @@
 
 #include "fwd_declare.h"
 
+#include <unordered_map>
+
```

Comment:
Do you still need this include? (Not sure if there's other code in the file that uses it, but looks like you removed two below